### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by Taoshix**
Comments

### CODE REVIEW

File: include/Route.hpp [hunk 1]
```diff
@@ -13,14 +13,23 @@ using namespace std;
...
 class Route {
  protected:
+    /// @brief Shared pointer to the database.
     shared_ptr<Database> db;
+
+    /// @brief Shared pointer to the webserver object.
     shared_ptr<Server> server;
+
+    /// @brief Shared pointer to the prediction manager.
     PredictionManager manager;
```
Comment (Mast3rwaf1z) [inline include/Route.hpp:29]: these are not particularly needed, its like ah yes, this getter gets the object

I think you can omit these comments

File: include/Route.hpp [hunk 2]
```diff
@@ -57,22 +74,35 @@ class Route {
         return false;
     }
 
+    /// @brief Responds to the request with json data and status code.
```
Comment (Mast3rwaf1z) [inline include/Route.hpp:77]:
```suggestion
    /// @brief Wrapper around response.status and response.set_content, used to determine content-type via overloading
```

File: include/Route.hpp [hunk 3]
```diff
@@ -57,22 +74,35 @@ class Route {
...
+    /// @param response
+    /// @param data
+    /// @param status
     void respond(Response* response, json data, int status = 200) {
         response->status = status;
         response->set_content(to_string(data), "application/json");
     }
 
+    /// @brief Responds to the request with plaintext data and status code.
```
Comment (Mast3rwaf1z) [inline include/Route.hpp:86]: can be omitted, or be the same as above

File: include/Routes/Answers.hpp [hunk 4]
```diff
@@ -8,10 +8,13 @@ using namespace httplib;
 using namespace nlohmann;
 
 class Answers : public Route {
+    // Inherits the super class constructor.
     using Route::Route;
 
  public:
+    /// @brief Adds the Answers endpoints.
```
Comment (Mast3rwaf1z) [inline include/Routes/Answers.hpp:15]:
```suggestion
    /// @brief Initializes the Answers endpoints.
```

File: include/Routes/Journals.hpp [hunk 5]
```diff
@@ -11,12 +11,19 @@ using namespace nlohmann;
 using namespace std;
 
 class Journals : public Route {
+    // Inherit the super class constructor.
     using Route::Route;
+
+    /// @brief Adds the Journal endpoints.
```
Comment (Mast3rwaf1z) [inline include/Routes/Journals.hpp:17]:
```suggestion
    /// @brief Initializes the Journal endpoints.
```

File: include/Routes/Journals.hpp [hunk 6]
```diff
@@ -56,6 +64,7 @@ class Journals : public Route {
                 db_int(jid));
             respond(&response, response_data);
         });
+        /// @brief Returns all the journals belonging to a specific user.
```
Comment (Mast3rwaf1z) [inline include/Routes/Journals.hpp:67]:
```suggestion
        /// @brief Returns all the journal ids belonging to a specific user.
```

File: include/Routes/Predictions.hpp [hunk 7]
```diff
@@ -11,10 +11,13 @@ using namespace std;
 using namespace nlohmann;
 
 class Predictions : public Route {
+    // Inherits the super class constructor.
     using Route::Route;
 
  public:
+    /// @brief Adds the Prediction endpoints.
```
Comment (Mast3rwaf1z) [inline include/Routes/Predictions.hpp:18]:
```suggestion
    /// @brief Initializes the Prediction endpoints.
```

File: include/Routes/Questions.hpp [hunk 8]
```diff
@@ -8,10 +8,13 @@ using namespace std;
 using namespace nlohmann;
 
 class Questions : public Route {
+    // Inherits the super class constructor.
     using Route::Route;
 
  public:
+    /// @brief Adds the Questions endpoints.
```
Comment (Mast3rwaf1z) [inline include/Routes/Questions.hpp:15]:
```suggestion
    /// @brief Initializes the Questions endpoints.
```

File: include/Routes/Settings.hpp [hunk 9]
```diff
@@ -10,10 +10,13 @@ using namespace httplib;
 using namespace nlohmann;
 
 class Settings : public Route {
+    // Inherits the super class constructor.
     using Route::Route;
 
  public:
+    /// @brief Adds the Settings endpoints.
```
Comment (Mast3rwaf1z) [inline include/Routes/Settings.hpp:17]:
```suggestion
    /// @brief Initializes the Settings endpoints.
```

File: include/Routes/Users.hpp [hunk 10]
```diff
@@ -11,18 +11,25 @@ using namespace httplib;
...
     return combinedhash;
 }
 
+/// @brief This class contains all of the endpoints related to users.
 class Users : public Route {
  public:
+    // Inherit the super class constructor.
     using Route::Route;
-
+    /// @brief Adds the User endpoints.
```
Comment (Mast3rwaf1z) [inline include/Routes/Users.hpp:30]:
```suggestion
    /// @brief Initializes the User endpoints.
```

File: include/Routes/Users.hpp [hunk 11]
```diff
@@ -124,6 +135,7 @@ class Users : public Route {
                 respond(&response, string("Username is already taken!"), 400);
             }
         });
+        /// @brief Updates a user's userdata with new values.
```
Comment (Mast3rwaf1z) [inline include/Routes/Users.hpp:138]:
```suggestion
        /// @brief Updates or creates a user's userdata with new values.
```

File: include/Table.hpp [hunk 12]
```diff
@@ -138,12 +142,15 @@ class Row {
...
 };
 
+/// @brief Data Structure for a table in the database.
 class Table {
  private:
     string name;
     vector<string> columns;
     shared_ptr<SQLite::Database> db;
 
+    /// @brief Executes a given SQL query and prints it if in verbose mode.
```
Comment (Mast3rwaf1z) [inline include/Table.hpp:152]:
```suggestion
    /// @brief Creates an SQL statement object with a given SQL query, abstracting database and printing
```

**Review by Mast3rwaf1z — CHANGES_REQUESTED** [review]
suggested a few changes

File: include/Route.hpp (hunk 3 above)
Comment (Taoshix) [reply include/Route.hpp:86]: changed to same as above

File: include/Route.hpp [hunk 13]
```diff
@@ -57,22 +69,35 @@ class Route {
...
+    /// @param data
+    /// @param status
     void respond(Response* response, string data, int status = 200) {
         response->status = status;
         response->set_content(data, "text/plain");
     }
 
 
  public:
+    /// @brief Initializes the http server and database.
```
Comment (Mast3rwaf1z) [inline include/Route.hpp:92]:
```suggestion
    /// @brief Constructs a route object with references to the http server and the database.
```

**Review by Mast3rwaf1z — CHANGES_REQUESTED** [review]
little change

**Pull request merged**